buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -1168,6 +1192,11 @@ public int setBytes(int index, ScatteringByteChannel in, int length) throws IOEx
             ByteBuf s = c.buf;
             int adjustment = c.offset;
             int localLength = Math.min(length, s.capacity() - (index - adjustment));
+            if (localLength == 0) {
```

Comment:
This fixes a bug for empty buffers in CompositeByteBuf. Without these lines, for empty buffers, `localReadBytes` will be 0, then it just breaks the loop and return. However, the correct way is trying the following buffers.